### PR TITLE
Add `videa` page (YouTube videos)

### PR DESCRIPTION
**Pull request by gabrielKerekes**
Not much to say about FE. The UI is basically copied (and adjusted) from post and/or podcast pages/components. I'll leave comments on non-obvious lines in the code.

BE PR: https://github.com/StreetOfCode/streetofcode-web-api/pull/17

### CODE REVIEW

File: src/pages/videa/[slug].tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+import Head from '../../components/Head'
+import {prefixWithHost, routes} from '../../routes'
+import styled from 'styled-components'
+import Flex from '../../components/core/Flex'
+import {device} from '../../theme/device'
+import AuthorAndDate from '../../components/domain/post/AuthorAndDate'
+import {youtubeVideoUrl, youtubeVideosUrl} from '../../api'
+import {YouTubeVideo} from '../../types'
+
+const videoDescriptionToHTML = (description: string) => {
```
Comment (gabrielKerekes) [inline src/pages/videa/[slug].tsx:14]: Some magic to make the video descriptions appear similarly to YouTube.  I.e. URLs become clickable links and the description lines are nicely separated.

File: src/components/domain/youtube/YouTubePreviewCard.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import {formatDate} from '../../../utils'
+import Flex from '../../core/Flex'
+import {device} from '../../../theme/device'
+import {YouTubeVideo} from '../../../types'
+
+type Props = {
+  video: YouTubeVideo
+}
+
+/***
```
Comment (xjahic) [inline src/components/domain/youtube/YouTubePreviewCard.tsx:15]: Edit comment pls

File: src/pages/videa/[slug].tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        </FlexWrapper>
+      </PageContentWrapper>
+    </>
+  )
+}
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const slug = context?.params?.slug as string
+
+  const video = await (await fetch(youtubeVideoUrl(slug))).json()
```
Comment (xjahic) [inline src/pages/videa/[slug].tsx:106]: Please add a logic to return notFound page if video with the specific slug doesn't exist. Now it just throws error

File: src/pages/videa/[slug].tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+                isPodcastOrVideo
+              />
+            </AuthorAndDateWrapper>
+          </AuthorAndTagsWrapper>
+          <PostContent
+            dangerouslySetInnerHTML={{
+              __html: videoDescriptionToHTML(video.snippet.description),
+            }}
+          />
+          <iframe
```
Comment (xjahic) [inline src/pages/videa/[slug].tsx:88]: This is not responsive. Tags bellow are not applied.

**Review by xjahic — CHANGES_REQUESTED** [review]
Otherwise LGTM. Search input would be nice as well.

File: src/components/domain/youtube/YouTubePreviewCard.tsx (hunk 2 above)
Comment (gabrielKerekes) [reply src/components/domain/youtube/YouTubePreviewCard.tsx]: https://github.com/StreetOfCode/streetofcode-web/pull/46/commits/65e6668df19e0f8c02f973574dc71ae95ff3c38c

File: src/pages/videa/[slug].tsx (hunk 3 above)
Comment (gabrielKerekes) [reply src/pages/videa/[slug].tsx]: https://github.com/StreetOfCode/streetofcode-web/pull/46/commits/65e6668df19e0f8c02f973574dc71ae95ff3c38c

File: src/pages/videa/[slug].tsx (hunk 4 above)
Comment (gabrielKerekes) [reply src/pages/videa/[slug].tsx:88]: moved the tags to a proper place. https://github.com/StreetOfCode/streetofcode-web/pull/46/commits/65e6668df19e0f8c02f973574dc71ae95ff3c38c

**Review by xjahic — APPROVED** [review]

**Pull request merged**